js/src/controls/LegendControl.js

```diff
@@ -0,0 +1,80 @@
+const L = require('../leaflet.js');
+const control = require('./Control.js')
+
+
+export class LeafletLegendControlModel extends control.LeafletControlModel {
+	defaults(){
+		return{
+			...super.defaults(),
+			_view_name : 'LeafletLegendControlView',
+			_model_name : 'LeafletLegendControlModel',
+			title: "Legend",
+			legend : {
+				"value 1": "#AAF",
+				"value 2": "#55A",
+				"value 3": "#005"
+			},
+			position: 'bottomright'
+		}
+	}
+}
+
+
+export class LeafletLegendControlView extends control.LeafletControlView{
+
+	initialize(parameters){
+		super.initialize(parameters);
+		this.map_view = this.options.map_view;
+	}
+
+	render(){
+		this.create_obj()
+		this.model.on('change:title', this.titleChanged, this);
+		this.model.on('change:position', this.positionChanged, this);
+		this.model.on('change:legend', this.legendChanged, this);
+	}
+
+	create_obj(title, positionning, legend){
+		let jsLegend = L.control({position: this.model.get('position')})
+
+		jsLegend.onAdd = (map) => {
+			let jsLegendName="leaflet-control-legend"
+			let container = L.DomUtil.create('div', jsLegendName)
+			this.addContent(container)
+			return container
+		}
+
+		this.obj = jsLegend
+		this.obj.addTo(this.map_view.obj)
```

Comment:
Just a final comment. This line should be removed. The map widget is responsible for automatically adding the layers/controls to the map. The only requirement is that the control is created under the `.obj` attribute, which is what you are doing already so it will work.

Reply:
Cool, I 've just done it :+1: